api/v1/serializers/players-serializer.js

```diff
@@ -76,7 +76,7 @@ const serializePlayer = (rawPlayers, gameId) => {
     identifierField: 'PLAYER_ID',
     resourceKeys: playerResourceKeys,
     resourcePath: playerResourcePathInstance,
-    topLevelSelfLink,
+    topLevelSelfLink: isPost ? playerResourceUrl : topLevelSelfLink,
```

Comment:
Instead of doing this logic here, it might be more clear if you move it to where you assign a value to the local variable `topLevelSelfLink` above.